tests/components/agency/AccountSpending-test.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { render, waitFor } from 'test-utils';
+import * as helpers from 'helpers/agencyV2Helper';
+import AccountSpending from 'components/agencyV2/accountSpending/AccountSpending';
+
+const defaultProps = {
+    agencyId: '012',
+    fy: '2020'
+};
+
+const mockData = {
+    data: []
+};
+
+let fetchBudgetaryResourcesSpy;
+let fetchSpendingCountSpy;
+let fetchSpendingByCategorySpy;
+
+beforeEach(() => {
+    fetchBudgetaryResourcesSpy = jest.spyOn(helpers, 'fetchBudgetaryResources').mockReturnValue({
+        promise: new Promise((resolve) => {
+            return process.nextTick(() => (resolve(mockData)));
+        }),
+        cancel: jest.fn()
+    }).mockClear();
+    fetchSpendingCountSpy = jest.spyOn(helpers, 'fetchSpendingCount').mockReturnValue({
+        promise: new Promise((resolve) => {
+            return process.nextTick(() => (resolve(mockData)));
+        }),
+        cancel: jest.fn()
+    }).mockClear();
+    fetchSpendingByCategorySpy = jest.spyOn(helpers, 'fetchSpendingByCategory').mockReturnValue({
+        promise: new Promise((resolve) => {
+            return process.nextTick(() => (resolve({
+                data: {
+                    page_metadata: {
+                        total: 1
+                    },
+                    results: []
+                }
+            })));
+        }),
+        cancel: jest.fn()
+    }).mockClear();
+});
+
+test('should fetch the data on first render ', () => {
+    render(<AccountSpending {...defaultProps} />);
+    return waitFor(() => {
+        expect(fetchBudgetaryResourcesSpy).toHaveBeenCalledTimes(1);
+        expect(fetchBudgetaryResourcesSpy).toHaveBeenCalledWith(defaultProps.agencyId, defaultProps.fy);
+        // expect(fetchSpendingCountSpy).toHaveBeenCalledTimes(1);
```

Comment:
@ebdabbs this is an existing bug... It is called like 4 times.

Reply:
Ok we can address that when we re-work this section!